pkg/plugin/installer/http_installer.go

```diff
@@ -64,7 +64,7 @@ var Extractors = map[string]Extractor{
 // This should be refactored in Helm 4, combined with the extension-based mechanism.
 func mediaTypeToExtension(mt string) (string, bool) {
 	switch strings.ToLower(mt) {
-	case "application/gzip", "application/x-gzip", "application/x-tgz", "application/x-gtar":
+	case "application/gzip", "application/x-gzip", "application/x-tgz", "application/x-gtar", "application/octet-stream":
```

Comment:
The addition of application/octet-stream doesn't mean it's a valid file. We need to make sure the extension is valid in the case of this. Can we pass it in to this function and then check for this case that the extension is right?